source/Halibut.Tests/WhenCancellingARequestForAPollingTentacle.cs

```diff
@@ -47,99 +46,201 @@ public class AndTheRequestHasBeenDequeuedButNoResponseReceived : BaseTest
+                responseMessages.Should().HaveCount(2);
+                responseMessages.ElementAt(0).Id.Should().Contain("IDoSomeActionService::ActionAsync");
+                responseMessages.ElementAt(0).Error.Should().NotBeNull();
+                responseMessages.ElementAt(0).Error!.Message.Should().Contain("The Request was cancelled while Transferring");
+                responseMessages.ElementAt(1).Error.Should().BeNull();
+                responseMessages.ElementAt(1).Id.Should().Contain("IEchoService::SayHelloAsync");
+
+                bool ShouldCancelOnDequeue()
+                {
+                    return shouldCancelWhenRequestDequeued;
+                }
+
+                void OnResponseApplied(ResponseMessage response)
+                {
+                    responseMessages.Add(response);
+                }
             }
+        }
 
-            [Test]
-            [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testListening: false, additionalParameters: new object[]{ false, true })]
-            [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testListening: false, additionalParameters: new object[]{ true, true })]
-            public async Task TheRequestShouldBeCancelled_WhenTheInProgressCancellationTokenIsCancelled_OnAsyncClients(
-                ClientAndServiceTestCase clientAndServiceTestCase, 
-                bool connectingCancellationTokenCancelled,
-                bool inProgressCancellationTokenCancelled)
-            {
-                var calls = new List<DateTime>();
-                var (tokenSourcesToCancel, halibutProxyRequestOptions) = CreateTokenSourceAndHalibutProxyRequestOptions(connectingCancellationTokenCancelled, inProgressCancellationTokenCancelled);
+        static (CancellationTokenSource TokenSourceToCancel, HalibutProxyRequestOptions HalibutProxyRequestOptions) CreateTokenSourceAndHalibutProxyRequestOptions()
+        {
+            var requestCancellationTokenSource = new CancellationTokenSource();
+            var halibutProxyRequestOptions = new HalibutProxyRequestOptions(requestCancellationTokenSource.Token);
+                
+            return (requestCancellationTokenSource, halibutProxyRequestOptions);
+        }
 
-                await using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
-                           .AsLatestClientAndLatestServiceBuilder()
-                           .WithHalibutLoggingLevel(LogLevel.Trace)
-                           .WithDoSomeActionService(() =>
-                           {
-                               calls.Add(DateTime.UtcNow);
+        /// <summary>
+        /// CancelWhenRequestQueuedPendingRequestQueueFactory cancels the cancellation token source when a request is queued
+        /// </summary>
+        class CancelWhenRequestQueuedPendingRequestQueueFactory : IPendingRequestQueueFactory
```

Comment:
It looks like this is putting back a class we moved into its own file before.

Reply:
Ahh good spot, it wasn't a pleasant rebase!!